nin/backend/serve.js

```diff
@@ -39,19 +39,61 @@ var serve = function(projectPath, shouldRunHeadlessly) {
       frontend.listen(8000);
     }
 
+    var findShaderDependencies = function(content) {
+      var regex = /\bSHADERS\.(\w+)\b/g;
+      var shaderNames = [];
+      var matches = content.match(regex) || [];
+      matches.forEach(function(match) {
+        var shaderName = match.split('.')[1];
+        if (shaderNames.indexOf(shaderName) === -1) {
+          shaderNames.push(shaderName);
+        }
+      });
+      return shaderNames;
+    };
+
+    var eventFromPath = function(data) {
+      var path = data.path,
+          filename = p.basename(path),
+          filenameWithoutExtension = filename.split('.')[0],
+          content = fs.readFileSync(path, 'utf-8');
```

Comment:
It's so visually unpleasing with all the var's though!
We can add a linter to the project though that can enforce whatever we might decide on regarding declarations.